docs(otel): document orphan-parent span drop behavior (#4485)
langchain-ai/docs: src/langsmith/trace-with-opentelemetry.mdx

* **Span ID**: A unique identifier for the current span
* **Sampling Decision**: Indicates whether this trace should be sampled

<Warning>
**A span whose parent is never sent to LangSmith is dropped.**

> The OTel endpoint is asynchronous: it accepts a batch, returns `200`, and materializes runs in the background. When a span's `parentSpanId` references a parent that LangSmith hasn't received yet, the child is buffered and linked once the parent arrives. This works regardless of order — a child can arrive before its parent, in a separate request, and still link correctly.

Reviewer: ```suggestion
The OTel endpoint is asynchronous: it accepts a batch, returns `200`, and materializes runs in the background. When a span's `parentSpanId` references a parent that LangSmith hasn't received yet, the child is buffered and linked once the parent arrives. This works regardless of order; a child can arrive before its parent, in a separate request, and still link correctly.
```

> However, if the parent span is **never** exported to LangSmith, the buffered child expires and never appears as a run — and no error is returned, since the `200` was sent before processing. This commonly happens when only part of a distributed trace reaches LangSmith: the parent is emitted by a service that exports to a different backend, or it's removed by a sampling decision.

Reviewer: ```suggestion
However, if the parent span is **never** exported to LangSmith, the buffered child expires and never appears as a run. LangSmith will not return an error, because the `200` was sent before processing. This commonly happens when only part of a distributed trace reaches LangSmith: the parent is emitted by a service that exports to a different backend, or it's removed by a sampling decision.
```

> To avoid silent loss, make sure every span you want in LangSmith has its ancestors exported to LangSmith as well, within the buffering window. On self-hosted deployments, this window is controlled by `REDIS_RUNS_EXPIRY_SECONDS` (default 12 hours).

Reviewer 2: What about SAAS?

> To avoid silent loss, make sure every span you want in LangSmith has its ancestors exported to LangSmith as well, within the buffering window. On self-hosted deployments, this window is controlled by `REDIS_RUNS_EXPIRY_SECONDS` (default 12 hours).

Reviewer: ```suggestion
To avoid silent loss, make sure every span you want in LangSmith also has its ancestors exported to LangSmith within the buffering window. On self-hosted deployments, this window is controlled by `REDIS_RUNS_EXPIRY_SECONDS` (default 12 hours).
```
